modules/salt/pages/salt-rate-limiting.adoc

```diff
@@ -0,0 +1,61 @@
+[[salt.rate.limiting]]
+= Rate Limiting
+
+Salt is able to run commands in parallel on a large number of minions.
+This can potentially create large amounts of load on your infrastructure.
+You can use these rate-limiting parameters to control the load in your environment.
+
+These parameters are all configured in the [filename]``/etc/rhn/rhn.conf`` configuration file.
+
+[NOTE]
+====
+Salt commands that are executed from the command line are not subject to these parameters.
+====
+
+
+
+== Batching
+
+There are two parameters that control how actions are sent to clients, one for the batch size, and one for the delay.
+
+When the Salt master sends a batch of actions to the target minions, it will send it to the number of minions determined in the batch size parameter.
+As each minion completes, the action begins on a new minion after the specified delay period.
+
+Choosing a lower batch size will reduce system load and parallelism, but might reduce overall performance for processing actions.
+
+The batch size parameter sets the maximum number of clients that can execute a single action at the same time.
+Adjust the [systemitem]``java.salt_batch_size`` parameter.
+Defaults to 100.
+
+Increasing the delay increases the chance that multiple minions will have completed before the next action is issued, resulting in fewer overall commands, and reducing load.
+
+The batch delay parameter sets the amount of time, in seconds, to wait after an action is processed before beginning to process the action on the next minion.
+Adjust the [systemitem]``java.salt_batch_delay`` parameter.
+Defaults to 1.0 seconds.
+
+
+
+== Presence Ping Timeout
+
+There are two parameters that control how presence pings from the Salt master are handled, one for the ping timeout, and one for the ping gather job.
+
+Salt batch calls begin with the Salt master performing a presence ping on the target minions.
+A ping gather job runs on the Salt master to handle the incoming pings from the minions.
+Batched actions will begin only after all minions have either responded to the ping, or timed out.
+
+The presence ping is an ordinary Salt command, so it is subject to the same timeout parameters as all other Salt commands.
+You can configure the global timeout values in the [filename]``/etc/salt/master.d/custom.conf`` configuration file.
+However, to allow for quicker detection of unresponsive minions, the timeout values for presence pings can be significantly shorter than those used elsewhere.
+Therefore, we recommend that you override the global timeout values by setting specific values for presence pings in [filename]``/etc/rhn/rhn.conf``.
```

Comment:
I think this needs to be clarified a bit more.

`salt_presence_ping_timeout` and `salt_presence_ping_gather_job_timeout` are *by default* set much lower than the global counterparts. We do not expect the majority of users to want to change these values.

By default, in `{productname}`:
* `timeout`: 2 minutes
* `gather_job_timeout`: 2 minutes
* total timeout for general Salt commands from `{productname}`: `timeout` + `gather_job_timeout` = 4 minutes
* `salt_presence_ping_timeout`: 1 second
* `salt_presence_ping_gather_job_timeout`: 4 seconds
* total timeout for presence pings: `salt_presence_ping_timeout` + `salt_presence_ping_gather_job_timeout` = 5 seconds

Users will rarely want to set those even lower.

The lower the total presence ping timeout is, the higher the chance for false negatives: minions reported as non-responding while they are actually responding, they might just be too busy (or too slow for whatever reason) to respond quickly enough.

The higher the total presence ping timeout is, the more accurate the test will be: most minions, even if slow, will have a chance to respond to the presence ping before timeouts elapse. OTOH, commands will start later and this could limit throughput when targeting a large set of minions some of which are slow.

Note that if all minions are fast, the choice of presence ping timeouts is irrelevant!

Reply:
OK, I've added that bit at the end. I can't really tell what changes you need in lines 47-50.

Reply:
After a call @moio 2019-05-14 I think we have this resolved now.